lib/collection/src/shard/replica_set.rs

```diff
@@ -142,6 +149,7 @@ impl ReplicaSet {
                 todo!("Add remote replica")
             }
         }
+        self.replica_state.save()?;
```

Comment:
I decided to simply make `save` public to perform a single atomic write instead of performing several writes within the loops.